src/main/java/ca/corefacility/bioinformatics/irida/model/workflow/submission/AnalysisSubmissionTempFile.java

```diff
@@ -0,0 +1,95 @@
+package ca.corefacility.bioinformatics.irida.model.workflow.submission;
+
+import java.nio.file.Path;
+import java.util.Date;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+import ca.corefacility.bioinformatics.irida.model.IridaThing;
+
+/**
+ * A temporary file which required by an {@link AnalysisSubmission} when
+ * the storage type is an object store.
+ */
+@Entity
+@Table(name = "analysis_submission_temp_files")
+public class AnalysisSubmissionTempFile implements IridaThing {
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private final Long id;
+
+	@NotNull
+	@Column(name = "analysis_submission_id")
+	private final Long analysisSubmissionId;
```

Comment:
Rather than storing the `analysisSubmissionId` can you link directly to the `AnalysisSubmission`?  That way we can keep some integrity between the objects and it's not just a loose ID being passed around.

Reply:
Linked directly to the AnalysisSubmission in 28c6030222586ca96886f9096c9e5cde3533bfd1